### PR TITLE
Fixed test build failures under gcc 9

### PR DESCRIPTION
**Pull request by dmitri-d**
Signed-off-by: Dmitri Dolguikh <ddolguik@redhat.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Commit Message: Fixed test build failures under gcc 9
Additional Description:
Risk Level: low
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: test/extensions/filters/http/common/fuzz/filter_fuzz_test.cc [hunk 1]
```diff
@@ -13,9 +13,9 @@ namespace Extensions {
 namespace HttpFilters {
 
 DEFINE_PROTO_FUZZER(const test::extensions::filters::http::FilterFuzzTestCase& input) {
-  static PostProcessorRegistration reg = {[](test::extensions::filters::http::FilterFuzzTestCase*
-                                                 input,
-                                             unsigned int seed) {
+  static PostProcessorRegistration reg __attribute__((
```
Comment (lizan) [inline test/extensions/filters/http/common/fuzz/filter_fuzz_test.cc:16]: use `ABSL_ATTRIBUTE_UNUSED` so it won't affect platforms doesn't support __attribute__? (likely MSVC)

Comment (dmitri-d) [reply test/extensions/filters/http/common/fuzz/filter_fuzz_test.cc]: done.

File: test/common/stats/stat_merger_fuzz_test.cc [hunk 2]
```diff
@@ -31,7 +31,7 @@ void testDynamicEncoding(absl::string_view data, SymbolTable& symbol_table) {
     // TODO(#10008): We should remove the "1 +" below, so we can get empty
     // segments, which trigger some inconsistent handling as described in that
     // bug.
-    uint32_t num_bytes = 1 + data[index] & 0x7;
+    uint32_t num_bytes = (1 + data[index]) & 0x7;
```
Comment (antoniovicente) [inline test/common/stats/stat_merger_fuzz_test.cc:34]: @jmarantz 

If you look at the comment above, I think the intent of this line was:
uint32_t num_bytes = 1 + (data[index] & 0x7);

Comment (twghu) [reply test/common/stats/stat_merger_fuzz_test.cc:34]: Checking...

**Pull request merged**